angularjs/angular-mocks.d.ts

```diff
@@ -97,137 +97,208 @@ declare module angular {
-
-        whenGET(url: string, headers?: Object): mock.IRequestHandler;
-        whenGET(url: string, headers?: (object: Object) => boolean): mock.IRequestHandler;
-        whenGET(url: RegExp, headers?: Object): mock.IRequestHandler;
-        whenGET(url: RegExp, headers?: (object: Object) => boolean): mock.IRequestHandler;
-
-        whenHEAD(url: string, headers?: Object): mock.IRequestHandler;
-        whenHEAD(url: string, headers?: (object: Object) => boolean): mock.IRequestHandler;
-        whenHEAD(url: RegExp, headers?: Object): mock.IRequestHandler;
-        whenHEAD(url: RegExp, headers?: (object: Object) => boolean): mock.IRequestHandler;
-
-        whenJSONP(url: string): mock.IRequestHandler;
-        whenJSONP(url: RegExp): mock.IRequestHandler;
-
-        whenPATCH(url: string, data?: string, headers?: Object): mock.IRequestHandler;
-        whenPATCH(url: string, data?: RegExp, headers?: Object): mock.IRequestHandler;
-        whenPATCH(url: string, data?: (data: string) => boolean, headers?: Object): mock.IRequestHandler;
-        whenPATCH(url: string, data?: Object, headers?: Object): mock.IRequestHandler;
-        whenPATCH(url: RegExp, data?: string, headers?: Object): mock.IRequestHandler;
-        whenPATCH(url: RegExp, data?: RegExp, headers?: Object): mock.IRequestHandler;
-        whenPATCH(url: RegExp, data?: (data: string) => boolean, headers?: Object): mock.IRequestHandler;
-        whenPATCH(url: RegExp, data?: Object, headers?: Object): mock.IRequestHandler;
-
-        whenPOST(url: string, data?: string, headers?: Object): mock.IRequestHandler;
-        whenPOST(url: string, data?: RegExp, headers?: Object): mock.IRequestHandler;
-        whenPOST(url: string, data?: (data: string) => boolean, headers?: Object): mock.IRequestHandler;
-        whenPOST(url: string, data?: Object, headers?: Object): mock.IRequestHandler;
-        whenPOST(url: RegExp, data?: string, headers?: Object): mock.IRequestHandler;
-        whenPOST(url: RegExp, data?: RegExp, headers?: Object): mock.IRequestHandler;
-        whenPOST(url: RegExp, data?: (data: string) => boolean, headers?: Object): mock.IRequestHandler;
-        whenPOST(url: RegExp, data?: Object, headers?: Object): mock.IRequestHandler;
-
-        whenPUT(url: string, data?: string, headers?: Object): mock.IRequestHandler;
-        whenPUT(url: string, data?: RegExp, headers?: Object): mock.IRequestHandler;
-        whenPUT(url: string, data?: (data: string) => boolean, headers?: Object): mock.IRequestHandler;
-        whenPUT(url: string, data?: Object, headers?: Object): mock.IRequestHandler;
-        whenPUT(url: RegExp, data?: string, headers?: Object): mock.IRequestHandler;
-        whenPUT(url: RegExp, data?: RegExp, headers?: Object): mock.IRequestHandler;
-        whenPUT(url: RegExp, data?: (data: string) => boolean, headers?: Object): mock.IRequestHandler;
-        whenPUT(url: RegExp, data?: Object, headers?: Object): mock.IRequestHandler;
+        /**

```

Comment:
My 16 union based signatures with documentation which replace the roughly 140 signatures necessary via overloading.